apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4CoverageResourceProvider.java

```diff
@@ -213,6 +213,11 @@ public Coverage read(@IdParam IdType coverageId) {
 
     Coverage coverage =
         coverageTransformer.transform(coverageIdSegment.get(), beneficiaryEntity, profileUsed);
+
+    if (coverage == null || coverage.isEmpty()) {
```

Comment:
Do we need the check for `isEmpty()` here? I'm not entirely sure what would cause that condition. Maybe it's safer to only check null here?

Reply:
I am not sure what would cause that either but I thought it would better to cover it. Doesn't it just add extra layer of safety if that ever happens?

Reply:
It looks like that method returns true if all of the properties are empty. I don't think we want to throw an exception in that case.

Reply:
updated